java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java

```diff
@@ -0,0 +1,100 @@
+package com.company.automation.classes.main_task;
+
+public class StudentLogic {
+    private final static StudentLogic studentLogic = new StudentLogic();
+
+    private StudentLogic() {
+    }
+
+    public static StudentLogic getStudentLogic() {
+        return studentLogic;
+    }
```

Comment:
I think that this way of realizing of singleton will be better =):

public class StudentLogic {
    private final static StudentLogic studentLogic;

    private StudentLogic() {
    }

    public static StudentLogic getStudentLogic() {
        if (studentLogic == null) {
            studentLogic = new StudentLogic();
        }
        return studentLogic;
    }
}

Reply:
Fixed